ui/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js

```diff
@@ -56,7 +57,8 @@ export default class PermissionPageContainerContent extends PureComponent {
       });
       return accumulator;
     }, []);
-
+    const { origin } = subjectMetadata;
+    const dappOrigin = isSnapId(origin) ? origin : getURLHost(origin);
```

Comment:
```suggestion
    const { origin, subjectType } = subjectMetadata;
    const displayOrigin = subjectType === SubjectType.Website ? getURLHost(origin) ? : origin;
```
How about we use the subject metadata for this? Something like this?

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/30092/commits/a0d34adefd30f919334a66861628acd1aedd2e1d